files/migrations.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/python3
+"""
+Runs migration scripts for this project
+
+Intended to be triggered by the %post rpm scriptlet rather than manually.
+"""
+
+import os
+import subprocess
+import sys
+from glob import glob
+
+VERSION_FILE = "/var/lib/{}/version"
+MIGRATIONS_DIR = "/usr/libexec/{}/migrations/"
+
+
+class Version:
```

Comment:
I'm not really super enthused about writing all these magic methods tbh. I would rather we have a function that normalizes a string version to a int tuple, e.g. `"1.2.3"` -> `(1,2,3,0)` which your `__init__` mostly does and then calling normal Python sort on that: `sorted(migrations, key=lambda m: m.version_as_tuple())` so we don't need to implement all the magic methods.

Reply:
I had no idea that tuples were compared/sorted this way, that's great, thanks!